nengo_loihi/tests/test_neurons.py

```diff
@@ -118,3 +169,231 @@ def test_lif_min_voltage(Simulator, plt, allclose):
+    ])
+def test_nengo_dl_noise(neuron_type, seed, plt, allclose):
+    nengo_dl = pytest.importorskip('nengo_dl')
+
+    dt = 0.001
+    nx = 256  # number of x points
+    n_noise = 500  # number of noise samples per x point
+
+    gain = 1
+    bias = 0
+
+    x = np.linspace(-1, 30, nx)
+
+    params = dict(amplitude=neuron_type.amplitude,
+                  tau_rc=neuron_type.tau_rc,
+                  tau_ref=neuron_type.tau_ref)
+    params2 = dict(params)
+    params2['tau_ref'] = params2['tau_ref'] + 0.5*dt
+
+    with nengo.Network() as model:
+        u = nengo.Node([0] * nx)
+        a = nengo.Ensemble(nx, 1, neuron_type=neuron_type,
+                           gain=nengo.dists.Choice([gain]),
+                           bias=nengo.dists.Choice([bias]))
+        nengo.Connection(u, a.neurons, synapse=None)
+        ap = nengo.Probe(a.neurons)
+
+    # --- compute rates
+    y_ref = loihi_rates(neuron_type, x, gain, bias, dt=dt)
+    y_med = nengo.LIF(**params2).rates(x, gain, bias)
+
+    with nengo_dl.Simulator(
+            model, dt=dt, minibatch_size=n_noise, seed=seed) as sim:
+        input_data = {u: np.tile(x[None, None, :], (n_noise, 1, 1))}
+        sim.step(input_feeds=input_data,
+                 extra_feeds={sim.tensor_graph.signals.training: True})
+        y = sim.data[ap][:, 0, :]
+
+    ymean = y.mean(axis=0)
+    y25 = np.percentile(y, 25, axis=0)
+    y75 = np.percentile(y, 75, axis=0)
+    dy25 = y25 - y_ref
+    dy75 = y75 - y_ref
+
+    # exponential models roughly fitted to 25/75th percentiles
+    x1mask = x > 1.1
+    x1 = x[x1mask]
+    if isinstance(neuron_type.nengo_dl_noise, AlphaRCNoise):
+        exp_model = 0.5 + 3.0*np.exp(-0.2*(x1 - 1))
+    elif isinstance(neuron_type.nengo_dl_noise, LowpassRCNoise):
+        exp_model = 1.5 + 2.2*np.exp(-0.3*(x1 - 1))
+
+    # --- plots
+    plt.subplot(211)
+    plt.plot(x, y_med, '--', label='LIF(tau_ref += 0.5*dt)')
+    plt.plot(x, ymean, label='nengo_dl')
+    plt.plot(x, y25, ':', label='25th')
+    plt.plot(x, y75, ':', label='75th')
+    plt.plot(x, y_ref, 'k--', label='LoihiLIF')
+    plt.legend()
+
+    plt.subplot(212)
+    plt.plot(x, ymean - y_ref, label='mean')
+    plt.plot(x, y25 - y_ref, ':', label='25th')
+    plt.plot(x, y75 - y_ref, ':', label='75th')
+    plt.plot(x1, exp_model, 'k--')
+    plt.plot(x1, -exp_model, 'k--')
+    plt.legend()
+
+    assert allclose(ymean, y_ref, atol=0.5)  # depends on n_noise
+    assert allclose(dy25[x1mask], -exp_model, atol=0.45, rtol=0.2)
+    assert allclose(dy75[x1mask], exp_model, atol=0.45, rtol=0.2)
+
+
+def test_no_nengo_dl(Simulator):
+    # check that things still work without nengo_dl
+    saved = neurons.nengo_dl, neurons.tf
+    neurons.nengo_dl = neurons.tf = None
+
+    with nengo.Network() as net:
+        a = nengo.Ensemble(10, 1, neuron_type=LoihiLIF())
+        nengo.Probe(a)
+
+    with Simulator(net) as sim:
+        sim.step()
+
+    neurons.nengo_dl, neurons.tf = saved
```

Comment:
Should we be putting this in a try/finally block or using mock, just to make sure if there's an error that these get reset and don't affect other tests?